### PR TITLE
[FABCAG-18] clearer error message when no public functions

### PR DESCRIPTION
**Pull request by awjh-ibm**
[FABCAG-18](https://jira.hyperledger.org/browse/FABCAG-18)

Add new check when adding a contract for if it has no transactions. Moves error from occuring due to metadata not validating against the schema to addContract function and thus gives clearer error

### CODE REVIEW

File: contractapi/contract_chaincode.go [hunk 1]
```diff
@@ -231,7 +231,7 @@ func (cc *ContractChaincode) addContract(contract ContractInterface, excludeFunc
 	}
 
 	if _, ok := cc.contracts[ns]; ok {
-		return fmt.Errorf("Multiple contracts being merged into chaincode with name %s", contract.GetName())
+		return fmt.Errorf("Multiple contracts being merged into chaincode with name %s", ns)
```
Comment (heatherlp) [inline contractapi/contract_chaincode.go]: why have you changed this here? you haven't changed it everywhere in this file..

File: contractapi/contract_chaincode.go [hunk 2]
```diff
@@ -311,6 +311,10 @@ func (cc *ContractChaincode) addContract(contract ContractInterface, excludeFunc
 		}
 	}
 
+	if len(ccn.functions) == 0 {
+		return fmt.Errorf("Contracts are required to have at least 1 chaincode callable method. Contract %s has none", ns)
```
Comment (heatherlp) [inline contractapi/contract_chaincode.go]: I'm not sold on this error message. what does "chaincode callable" mean? I might think my private method is "chaincode callable"? if I knew what it meant ;)

File: contractapi/contract_chaincode_test.go [hunk 3]
```diff
@@ -438,6 +446,13 @@ func TestAddContract(t *testing.T) {
 	err = cc.addContract(mc, []string{})
 	assert.EqualError(t, err, "Multiple contracts being merged into chaincode with name customname", "should error when contract already exists with name")
 
+	// should error when no public functions
+	cc = new(ContractChaincode)
+	cc.contracts = make(map[string]contractChaincodeContract)
+	pc := new(privateContract)
+	err = cc.addContract(pc, defaultExcludes)
+	assert.EqualError(t, err, "Contracts are required to have at least 1 chaincode callable method. Contract privateContract has none", "should error when contract has no public functions")
```
Comment (heatherlp) [inline contractapi/contract_chaincode_test.go]: would you expect it to behave differently if the contract just had no methods in it? as opposed to just private ones?

File: contractapi/contract_chaincode.go (hunk 1 above)
Comment (awjh-ibm) [reply contractapi/contract_chaincode.go]: The if above works out what to call the contract if GetName returns blank I realized when adding the new error message that the error was using the wrong value and therefore would be blank. Unlikely to hit it as a problem as go would prevent you having two types with the same name in the same package but better for neatness

File: contractapi/contract_chaincode.go (hunk 2 above)
Comment (awjh-ibm) [reply contractapi/contract_chaincode.go]: Agreed

**Pull request merged**